src/OrleansGetEventStore/GetEventStoreProvider.cs

```diff
@@ -0,0 +1,304 @@
+                    return new EventStreamSegment()
+                    {
+                        StreamName = streamName,
+                        FromVersion = 0,
+                        ToVersion = 0,
+                        Events = emptyList
+                    };
+
+                if (fromVersion > currentSlice.LastEventNumber)
+                {
+                    // the fromVersion is larger than last event in the slice; 
+                    // this means the query targeted a non-existing segment. Return empty segment.
+                    return new EventStreamSegment()
+                    {
+                        StreamName = streamName,
+                        FromVersion = currentSlice.LastEventNumber + 1,
+                        ToVersion = currentSlice.LastEventNumber + 1,
+                        Events = emptyList
+                    };
+                }
+
+                foreach (var evnt in currentSlice.Events)
+                      events.Add(new KeyValuePair<Guid,object>(evnt.Event.EventId, DeserializeEvent(evnt.Event)));
+
+                sliceStart = currentSlice.NextEventNumber;
+
+            } while (!currentSlice.IsEndOfStream);
+
+            return new EventStreamSegment()
+            {
+                StreamName = streamName,
+                FromVersion = fromVersion,
+                ToVersion = fromVersion + events.Count,
+                Events = events
+            };
+        }
+
+        public async Task<bool> Delete(string streamName, int? expectedVersion)
+        {
+            try
+            {
+                var versionParameter = expectedVersion.HasValue ? expectedVersion.Value - 1 : ExpectedVersion.Any;
+                var result = await this.Connection.DeleteStreamAsync(streamName, versionParameter, true);
+                return true;
+            }
+            catch (WrongExpectedVersionException)
+            {
+                return false;
+            }
+            catch (StreamDeletedException)
+            {
+                return false;
+            }
+        }
+
+        public static object DeserializeEvent(RecordedEvent @event)
+        {
+            // Backwards compatibility
+            var metadata = DeserializeMetadata(@event.Metadata);
+
+            var eventTypeProperty = metadata.Property("EventClrTypeName");
+
+            if (eventTypeProperty != null)
+                return DeserializeEvent((string)eventTypeProperty.Value, @event.Data);
+            // Backwards compatibility
+
+            return DeserializeEvent(@event.EventType, @event.Data);
+        }
+
+        public static JObject DeserializeMetadata(byte[] metadata)
+        {
+            return JObject.Parse(Encoding.UTF8.GetString(metadata));
+        }
+
+        public static object DeserializeEvent(string eventClrTypeName, byte[] data)
+        {
+            return JsonConvert.DeserializeObject(Encoding.UTF8.GetString(data), Type.GetType(eventClrTypeName));
```

Comment:
I will copy the serialization configuration mechanism used in the AzureBlobStorage provider.